### PR TITLE
implement affected_count for sqlite backend

### PR DESCRIPTION
**Pull request by jabot2**
This PR tries to implement the affected_count method for the sqlite backend.

### CODE REVIEW

File: lib-driver/caqti_driver_sqlite3.ml [hunk 1]
```diff
@@ -299,16 +299,33 @@ module Connect_functor (System : Caqti_driver_sig.System_unix) = struct
...
+          return (Error (Caqti_error.response_rejected ~uri ~query msg))
+
+      let run_step response =
+        let ret = Sqlite3.step response.stmt in
+        if response.has_been_executed
+        then ret
+        else
+          (response.has_been_executed <- true;
+           response.affected_count <- Sqlite3.changes db;
+           ret)
```
Comment (paurkedal) [inline lib-driver/caqti_driver_sqlite3.ml:324]:
```suggestion
        if not response.has_been_executed then
          (response.has_been_executed <- true;
           response.affected_count <- Sqlite3.changes db);
        ret
```

File: lib-driver/caqti_driver_sqlite3.ml [hunk 2]
```diff
@@ -299,16 +299,33 @@ module Connect_functor (System : Caqti_driver_sig.System_unix) = struct
...
+
+      let fetch_row response =
+        let {stmt; row_type; query; _} = response in
```
Comment (paurkedal) [inline lib-driver/caqti_driver_sqlite3.ml:327]: Please use `response.*` notation below instead, or keep the original pattern match and alias with `as`.

File: lib-driver/caqti_driver_sqlite3.ml [hunk 3]
```diff
@@ -324,10 +341,11 @@ module Connect_functor (System : Caqti_driver_sig.System_unix) = struct
          | rc ->
             Error (Caqti_error.response_failed ~uri ~query (Rc rc)))
 
-      let exec {stmt; row_type; query} =
+      let exec response =
+        let {row_type; query; _} = response in
```
Comment (paurkedal) [inline lib-driver/caqti_driver_sqlite3.ml:345]: See above.

**Comment by paurkedal** [comment]
This is very useful. Should be ready to merge, but please take a look at my suggestions first.

File: lib-driver/caqti_driver_sqlite3.ml (hunk 1 above)
Comment (jabot2) [reply lib-driver/caqti_driver_sqlite3.ml:324]: will do

File: lib-driver/caqti_driver_sqlite3.ml (hunk 2 above)
Comment (jabot2) [reply lib-driver/caqti_driver_sqlite3.ml:327]: ok

File: lib-driver/caqti_driver_sqlite3.ml (hunk 3 above)
Comment (jabot2) [reply lib-driver/caqti_driver_sqlite3.ml:345]: ok

**Comment by paurkedal** [comment]
Thanks!

**Pull request merged**